core/trino-main/src/main/java/io/trino/server/ui/OAuth2WebUiAuthenticationFilter.java

```diff
@@ -150,4 +159,9 @@ private boolean hasValidAudience(Object audience)
         }
         return false;
     }
+
+    private boolean isValidPrincipal(Object principal)
+    {
+        return principal instanceof String && !Strings.isNullOrEmpty((String) principal);
```

Comment:
`instanceof` also verifies that `principal != null` so I think you do:
```
        return principal instanceof String && !principal.isEmpty();
```

Reply:
Of course! I took @lukasz-walkiewicz's suggestion a little too literally! 😄 Fixed and squashed.